cpp/CMakeLists.txt

```diff
@@ -162,9 +162,6 @@ if("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_SOURCE_DIR}")
     "Build with zstd compression"
     ON)
 
-  option(ARROW_ZLIB_VENDORED
-    "Build our own zlib (some libz.a aren't configured for static linking)"
-    ON)
```

Comment:
@wesm this variable seems is not really needed, since we build all others 3rd party libs from sources by default too. But it brings difference and misunderstanding between ways how arrow cmake script works with 3rd party libs. I have provided some more details below regarding this.